src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/projectpage/model/ProjectPageModel.kt

```diff
@@ -0,0 +1,147 @@
+    var projectTitle: String by property()
+    val projectTitleProperty = getProperty(ProjectPageModel::projectTitle)
+
+    // List of collection children (i.e. the chapters) to display in the list
+    var children: ObservableList<Collection> = FXCollections.observableList(mutableListOf())
+
+    // Selected child
+    var activeChild: Collection by property()
+    val activeChildProperty = getProperty(ProjectPageModel::activeChild)
+
+    // List of chunks to display on the screen
+    var chunks: ObservableList<Chunk> = FXCollections.observableList(mutableListOf())
+
+    var activeChunk: Chunk by property()
+    var activeChunkProperty = getProperty(ProjectPageModel::activeChunk)
+
+    // What record/review/edit context are we in?
+    var context: ChapterContext by property(ChapterContext.RECORD)
+    var contextProperty = getProperty(ProjectPageModel::context)
+
+    // Whether the UI should show the plugin as active
+    var showPluginActive: Boolean by property(false)
+    var showPluginActiveProperty = getProperty(ProjectPageModel::showPluginActive)
+
+    // Keep a view context to start transitions
+    var workspace: Workspace? = null
+
+    val projectPageActions = ProjectPageActions(
+            Injector.directoryProvider,
+            WaveFileCreator(),
+            Injector.collectionRepo,
+            Injector.chunkRepository,
+            Injector.takeRepository,
+            Injector.pluginRepository
+    )
+
+    init {
+        // TODO: Get from scope (passed from home) instead of first from repo
+        Injector
+                .projectRepo
+                .getAllRoot()
+                .observeOnFx()
+                .subscribe { retrieved ->
+                    initializeView(retrieved.first())
+                }
+    }
+
+    fun initializeView(newProject: Collection) {
+        project = newProject
+        projectTitle = newProject.titleKey
+        projectPageActions
+                .getChildren(newProject)
+                .observeOnFx()
+                .subscribe { childCollections ->
+                    // Now we have the children of the project collection
+                    children.clear()
+                    children.addAll(childCollections)
+                }
+    }
+
+    fun selectChildCollection(child: Collection) {
+        activeChild = child
+        // Remove existing chunks so the user knows they are outdated
+        chunks.clear()
+        projectPageActions
+                .getChunks(child)
+                .observeOnFx()
+                .subscribe { retrieved ->
+                    chunks.clear() // Make sure any chunks that might have been added are removed
+                    chunks.addAll(retrieved)
+                }
+    }
+
+    fun checkIfChunkHasTakes(chunk: Chunk): Single<Boolean> {
+        return projectPageActions
+                .getTakeCount(chunk)
+                .map { it > 0 }
+    }
+
+    fun doChunkContextualAction(chunk: Chunk) {
```

Comment:
this functions job should be to delegate; let each switch condition be a separate function